sdk/src/Core/RegionEndpoint/RegionEndpointProviderV3.cs

```diff
@@ -348,12 +356,12 @@ class ServiceMap
             /// <summary>
             /// Stores the plain endpoints for each service in the current region
             /// </summary>
-            private Dictionary<string, RegionEndpoint.Endpoint> _serviceMap = new Dictionary<string, RegionEndpoint.Endpoint>();
+            private ConcurrentDictionary<string, RegionEndpoint.Endpoint> _serviceMap = new ConcurrentDictionary<string, RegionEndpoint.Endpoint>();
```

Comment:
Thanks for taking a stab at this! We made an attempt internally but haven't been able to figure it out yet.

Unfortunately we cannot rely on `ConcurrentDictionary` entirely yet as its not available in the .NET Framework 3.5 target of the SDK.
* Generally we prefer to avoid branching with preprocessor directives to reduce complexity, though given the reports in #1939 appear to focus on .NET 6 maybe it'd be acceptable if net35 isn't impacted as severely. Might need @normj to weigh in here.

Our thinking was that we'd need to focus on the `_serviceMap` lock up on line 97.

Reply:
Hi, Thank you for considering my changes. I read what you wrote and I'm sorry about the branch name, hope it didn't create too much mess. Regarding the `ConcurrentDictionary `, I now removed any use of it and just added `lock` instead. 
I do think the critical change is in line 97 as you mentioned but I added this behavior also in the class itself just to be on the safe side. I can change this how ever you like just lmk.

Reply:
Oh and BTW, our product uses a lot of AWS .net SDK packages and when we upgraded the AWSSDK.Core (and deployed to production) we got a lot of exception (not in stage environment) and we saw those exception from all around the SDK's referring to the same source, for example:
![image](https://user-images.githubusercontent.com/88372444/190889617-7bb10b20-2d7f-4868-b355-3c2f7f627195.png)